### PR TITLE
enhance relationship table with search functionality

### PR DESCRIPTION
**Pull request by wipascal**

### CODE REVIEW

**Comment by wipascal** [comment]
Now ready for review @saemideluxe

File: basxconnect/core/views/person/person_search_views.py [hunk 1]
```diff
@@ -30,7 +30,54 @@ def searchperson(request):
...
+            style="margin-bottom: 1rem; padding: 16px 0 48px 48px; background-color: #fff",
+        ).render({})
+    )
+
+
+# Search view
+# simple person search view, for use with ajax calls
+def searchperson_and_insert(request):
+    query = request.GET.get("q")
+    selected_result_selector = request.GET.get("target-id-to-store-selected")
```
Comment (saemideluxe) [inline basxconnect/core/views/person/person_search_views.py:60]: I think passing a selector via GET query is not an ideal solution. But changing that would be too much work for now I think, we can still do some refactoring when we do the next iteration.

**Review by saemideluxe — CHANGES_REQUESTED** [review]
Will be resolved together with the according PR in bread

File: basxconnect/core/views/person/person_search_views.py [hunk 2]
```diff
@@ -30,7 +30,54 @@ def searchperson(request):
...
+    objects = (
+        SearchQuerySet()
+        .models(models.Person)
+        .autocomplete(name_auto=query)
+        .filter_or(personnumber=query)
+    )
+
+    def onclick(person):
+        return (
+            f"set_value('#{selected_result_selector}', '{person.pk}');"
```
[the remark this answers is not in the document]
Comment (wipascal) [reply basxconnect/core/views/person/person_search_views.py]: it does

File: basxconnect/core/views/person/person_search_views.py [hunk 3]
```diff
@@ -40,57 +87,35 @@ def searchperson(request):
...
+            ),
+            " ",
+            mark_safe(highlight.highlight(person.search_index_snippet())),
+            style="cursor: pointer; padding: 8px 0;",
+            onclick=onclick(person),
+            onmouseenter="this.style.backgroundColor = 'lightgray'",
+            onmouseleave="this.style.backgroundColor = 'initial'",
+        )
+
+    result_list = list(filter(lambda x: x, map(_display_as_list_item, first_results)))
```
[the remark this answers is not in the document]
Comment (wipascal) [reply basxconnect/core/views/person/person_search_views.py]: done

File: basxconnect/core/views/person/person_search_views.py [hunk 4]
```diff
@@ -40,57 +87,35 @@ def searchperson(request):
     )
 
 
-def display_results(objects, highlight):
+def _display_results(objects, highlight, onclick):
     if objects.count() == 0:
         return _("No results")
```
[the remark this answers is not in the document]
Comment (wipascal) [reply basxconnect/core/views/person/person_search_views.py:92]: solved

File: basxconnect/core/views/person/person_search_views.py (hunk 1 above)
Comment (wipascal) [reply basxconnect/core/views/person/person_search_views.py:60]: will take care of it. resolving this conversation as we have another one open regarding the bigger issue.

**Comment by wipascal** [comment]
@saemideluxe PR is ready for a second review

**Review by saemideluxe — APPROVED** [review]
Okay, looks good.

**Pull request merged**